pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -612,18 +611,46 @@ func (r *hibernationReconciler) checkClusterRunning(cd *hivev1.ClusterDeployment
 		if err := r.updateClusterDeploymentStatus(cd, logger); err != nil {
 			return reconcile.Result{}, err
 		}
-		poolNS, poolName := "", ""
-		if cd.Spec.ClusterPoolRef != nil {
-			poolNS = cd.Spec.ClusterPoolRef.Namespace
-			poolName = cd.Spec.ClusterPoolRef.PoolName
-		}
 		// logging with time since hibernating condition was set to ResumingOrRunning state
-		hivemetrics.MetricClusterReadyTransitionSeconds.WithLabelValues(cd.Labels[constants.VersionMajorMinorPatchLabel],
-			cd.Labels[hivev1.HiveClusterPlatformLabel], poolNS, poolName).Observe(time.Since(hibernatingCondition.LastTransitionTime.Time).Seconds())
+		logMetric(hivemetrics.MetricClusterReadyTransitionSeconds, cd,
+			time.Since(hibernatingCondition.LastTransitionTime.Time).Seconds())
+		// Clear entry from currently resuming clusters if exists
+		deleteMetric(hivemetrics.MetricResumingClustersSeconds, cd)
 	}
 	return reconcile.Result{}, nil
 }
 
+// deleteMetric has the knowledge of labels needed to clear the metrics for currently transitioning clusters
+// Note: do not use for cumulative metrics - they have different labels and are not designed to be cleared
+func deleteMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment) {
```

Comment:
At a glance, `deleteMetric` and `logMetric` appear closely related. Reading the comments, I see that the former is intended for "transitioning" metrics and the latter for "cumulative" metrics. That being the case, suggest including that in the names to make it clearer.
```suggestion
func deleteTransitionMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment) {
```